pycalphad/io/cs_dat.py

```diff
@@ -1159,6 +1162,999 @@ def parse_cs_dat(instring):
+
+            # Write quadruplet coordination numbers for MQM
+            if phase_model in ('SUBG', 'SUBQ'):
+                detect_query = (
+                    (where("phase_name") == phase_name) & \
+                    (where("parameter_type") == "MQMZ")
+                )
+                params = dbf._parameters.search(detect_query)
+                for param in params:
+                    con = []
+                    for ion in param['constituent_array']:
+                        for species in ion:
+                            con.append(flat_constituents.index(species.name) + 1)
+                    con_string = ''.join([f"{c:4}" for c in con])
+                    z_string = '      '.join([f"{z:.7f}" for z in param['coordinations']])
+                    output += f'{con_string}  {z_string}\n'
+
+        # Write magnetic excess mixing data
+        if phase_model in ('RKMPM', 'SUBLM'):
+            # Get excess magnetic parameters
+            for tc in tcs:
+                tc_value = tc['parameter']
+                tc_constituents = tc['constituent_array']
+                tcs.remove(tc)
+                bmagn_value = 0
+                for bmagn in bmagns:
+                    # Look for matching bmagn
+                    if bmagn['constituent_array'] != tc_constituents:
+                        continue
+                    bmagn_value = bmagn['parameter']
+                    bmagns.remove(bmagn)
+                    break
+
+                # Index calculation depends on model
+                if   phase_model == 'RKMPM':
+                    # Get indices of participating constituents in phase (order of printed endmembers)
+                    indices = []
+                    for species in constituent_set:
+                        name = ''
+                        for element in species.constituents:
+                            # Add element name to endmember name if not vacancy
+                            if element.capitalize() != 'Va':
+                                name += element.capitalize()
+                                stoich = species.constituents[element] * weight
+                                # Add stoichiometric factor to name if not 1
+                                if stoich != 1:
+                                    if stoich % 1 == 0:
+                                        name += f'{int(stoich)}'
+                                    else:
+                                        name += f'{stoich:.2g}'
+                        try:
+                            indices.append(1 + endmember_names.index(name))
+                        except ValueError:
+                            print(f'Endmember {name} not found in phase {phase_name}')
+                            print(f'List of known endmembers: {endmember_names}')
+                            print('Write aborted')
+                            return
```

Comment:
This should call `raise ValueError` again with the error message so the exception is propagated correctly to the caller.